Lokad.AzureEventStore.Cache/AzureCacheProvider.cs

```diff
@@ -53,7 +53,11 @@ public AzureCacheProvider(CloudBlobContainer container)
         /// </remarks>
         private async Task<CloudBlockBlob[]> Blobs(string stateName)
         {
-            await _container.CreateIfNotExistsAsync();
+            // Attempting to create the container if it doesn't exist allows
+            // to jump-start a state from existing caches only letting the
+            // upload of a refreshed state to fail.
+            if (AllowWrite && ! await _container.ExistsAsync().ConfigureAwait(false))
```

Comment:
I'm not sure I understand what the comment means. What does 'jump-start a state' mean ? What does 'refreshed state' mean ? What does the 'only letting ...' sentence refer to ? Are the 'existing caches only letting' ? Is it 'jump-start .. only letting' ? 

There is also the matter of documenting why `ExistsAsync()` is called before `CreateIfNotExistsAsync()` since (judging by the name of the function) `CreateIfNotExistsAsync()` should only create the container if it does not exist, making the previous call apparently redundant.

Reply:
The new comment is better, but does not explain why `_container.ExistsAsync()` is called. 

The key piece of information, which is really not obvious, is that with read-only permissions, `_container.CreateIfNotExistsAsync()` will throw an exception **even if the container already exists**. This is very surprising behavior, and the reason we need to call `ExistsAsync()` beforehand.